amqp/src/main/java/io/github/tcdl/msb/adapters/amqp/AmqpAcknowledgementHandler.java

```diff
@@ -0,0 +1,77 @@
+package io.github.tcdl.msb.adapters.amqp;
+
+import io.github.tcdl.msb.api.AcknowledgementHandler;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.rabbitmq.client.Channel;
+
+/**
+ * {@link AmqpAcknowledgementHandler} provides acknowledgement for AMQP broker. 
+ * Used from {@link AmqpMessageConsumer}.
+ */
+
+public class AmqpAcknowledgementHandler implements AcknowledgementHandler {
+
+    private static final String ACK_WAS_ALREADY_SENT = "Acknowledgement was already sent during message processing.";
+
+    private static final Logger LOG = LoggerFactory.getLogger(AmqpAcknowledgementHandler.class);
+
+    final Channel channel;
+    final String consumerTag;
+    final long deliveryTag;
+    final boolean isRequeueRejectedMessages;
+
+    boolean isAcknowledgementSent = false;
+
+    public AmqpAcknowledgementHandler(Channel channel, String consumerTag, long deliveryTag,
+            boolean isRequeueRejectedMessages) {
+        super();
+        this.channel = channel;
+        this.consumerTag = consumerTag;
+        this.deliveryTag = deliveryTag;
+        this.isRequeueRejectedMessages = isRequeueRejectedMessages;
+    }
+
+    @Override
+    public void confirmMessage() {
+        if (!isAcknowledgementSent) {
```

Comment:
This method is not thread safe which may lead to problems with concurrent acknowldegements. If this class is mean to be non-thread-safe it must be specified in javadoc for `AcknowledgementHandler` because it's client-facing API.